lib/src/metta/runner/mod.rs

```diff
@@ -345,7 +345,13 @@ impl<'a> RunnerState<'a> {
 
             // Get the next atom, and start a new intperpreter
             let next_atom = if let Some(parser) = self.parser.as_mut() {
-                parser.parse(&self.metta.0.tokenizer.borrow())?
+                match parser.parse(&self.metta.0.tokenizer.borrow()) {
+                    Ok(atom) => atom,
+                    Err(err) => {
+                        self.mode = MettaRunnerMode::TERMINATE;
```

Comment:
Do we really need setting mode to `TERMINATE`? Looks like previous line should work nice.

Reply:
Do you see a reason that terminating the RunnerState on a parse error would be bad?

The reason I did it this way was because a parse error essentially results in a crash for that RunnerState (but not for the whole runner).  In other words, I don't see a situation where we'd want to resume execution in the same runner state after some MeTTa code failed to parse.  Potentially that could lead to very hard-to-debug situations where some code was silently skipped causing problems downstream.

In Rust it's a non-issue because Rust passes back `Result` types, which force the caller to consider errors.  But in the C API, it seemed likely code like this would be common:
```C
    while (!runner_state_is_complete(&runner_state))
        runner_state_step(&runner_state);
```
And I didn't want to create an API that would lead to infinite loop if it was slightly misused, and I didn't see a reason not to terminate the RunnerState on a syntax error.

Reply:
Thanks for explaining this. I asked mainly because there is another line above with error processing which doesn't set `TERMINATE` mode on `RunnerState`, and I was curious what is a difference:
```rs
                match interpreter_state.into_result() {
                    Err(msg) => return Err(msg),
```
Should we change it to terminate runner as well? Probably we will never got error in this snipped though.

Reply:
Good point.  That case should never be hit, but I simplified the code to make that clear in 53fd2ab.

Thanks!